pkg/limits/limit.go

```diff
@@ -46,6 +47,10 @@ func newLimitsChecker(limits Limits, store *usageStore, producer *producer, part
 			Name: "loki_ingest_limits_tenant_ingested_bytes_total",
 			Help: "Total number of bytes ingested per tenant within the active window. This is not a global total, as tenants can be sharded over multiple pods.",
 		}, []string{"tenant"}),
+		streamInvalidPartitionAssigned: promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
+			Name: "loki_ingest_limits_stream_invalid_partition_assigned_total",
```

Comment:
Similar suggestion, using plural naming convention for counters:

```suggestion
			Name: "loki_ingest_limits_streams_discarded_total",
```

and then we have `loki_ingest_limits_records_discarded_total` for consumption, and `streams_discarded_total` for the synchronous write path.